src/command_info.rs

```diff
@@ -40,7 +40,7 @@ pub fn run_command_info(paths: &GlobalPaths) -> Result<()> {
     let dbversion_url = juliaupserver_base.join(dbversion_url_path)
         .with_context(|| format!("Failed to construct a valid url from '{}' and '{}'.", juliaupserver_base, dbversion_url_path))?;
 
-    let online_dbversion = download_juliaup_version(&dbversion_url.to_string())
+    let online_dbversion = download_juliaup_version(dbversion_url.as_ref())
```

Comment:
Various places, I've avoided cloning, or copying a `str` to a `String`, if not needed.